sdk/aqueduct/decorator.py

```diff
@@ -373,15 +407,25 @@ def op(
 
         >>> recommendations.get()
     """
-    _type_check_decorator_arguments(
-        description,
-        file_dependencies,
-        requirements,
+    # Establish parity between `num_outputs` and `outputs`, or raise exception if there is a mismatch.
+    if num_outputs is None and outputs is None:
+        num_outputs = 1
+    elif num_outputs is not None and outputs is not None:
+        if len(outputs) != num_outputs:
+            raise InvalidUserArgumentException(
+                "`len(outputs) must be equivalent to `num_outputs`. Getting %d and %d"
+                % (len(outputs), num_outputs)
+            )
+    elif num_outputs is None and outputs is not None:
+        num_outputs = len(outputs)
```

Comment:
Will there be any issue if len(outputs) != actual num_outputs?

Reply:
yep, it'll fail at execution time.